pentasharp/Services/ReviewService.cs

```diff
@@ -0,0 +1,40 @@
+﻿using pentasharp.Data;
+using pentasharp.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+using pentasharp.Interfaces;
+
+namespace pentasharp.Services
+{
+    public class ReviewService : IReviewService
+    {
+        private readonly AppDbContext _context;
+
+        public ReviewService(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<bool> SubmitReviewAsync(UserReview review, int? userId)
+        {
+            if (review == null)
+                return false;
+
+            string userName = "Anonymous";
+
+            if (userId.HasValue)
+            {
+                var user = await _context.Users.FirstOrDefaultAsync(u => u.UserId == userId.Value);
+                userName = user?.FirstName ?? userName;
+                review.UserId = userId.Value;
```

Comment:
to avoid checking user here and further down, improve this code block to something like 
            if (userId.HasValue)
            {
                var user = await _context.Users
                    .AsNoTracking()
                    .FirstOrDefaultAsync(u => u.UserId == userId.Value);

                if (user != null)
                {
                    userName = user.FirstName;
                    review.UserId = userId.Value;
                }
            }

            review.UserName ??= userName;